lib/active_merchant/billing/gateways/merchant_e_solutions.rb

```diff
@@ -55,10 +57,11 @@ def capture(money, transaction_id, options = {})
       end
 
       def store(creditcard, options = {})
-        post = {}
-        post[:client_reference_number] = options[:customer] if options.has_key?(:customer)
-        add_creditcard(post, creditcard, options)
-        commit('T', nil, post)
+        response = MultiResponse.run do |r|
+          r.process { temporary_store(creditcard, options) }
+          r.process { verify(creditcard, { store_card: 'y' }) }
```

Comment:
I think the idea here is that `temporary_store` will create a temporary token at the gateway, which will be the `transaction_id` in the gateway's response.

THEN we need to grab the temporary token from the store response and pass it in the `verify` call, instead of `creditcard`. The result is that the gateway will accept the temporary token and return a permanent token in `card_id` field.

Reply:
Thanks, I updated the method to pass `r.authorization`.